File: app/src/main/java/com/yalonso/interview/ProductsActivity.java
package com.yalonso.interview;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import com.yalonso.interview.coreFragments.ProducDetailsFragment;
import com.yalonso.interview.coreFragments.ProductsFragment;
import com.yalonso.interview.models.Product;

public class ProductsActivity extends AppCompatActivity {

    ProductsFragment productsFragment;
    ProducDetailsFragment productsDetailsFragment;
    FragmentManager fragmentManager;
    FragmentTransaction fragmentTransaction;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        // setContentView();

        pushFragment();
    }

    protected void pushFragment() {
        productsFragment = ProductsFragment.newInstance();

        fragmentManager = getSupportFragmentManager();
        fragmentManager.beginTransaction()
                .replace(R.id.frame_body, productsFragment)
                .addToBackStack(ProductsFragment.TAG)
                .commit();
    }

    public void pushProductFragment(Product product) {
        productsDetailsFragment = ProducDetailsFragment.newInstance(product);

        fragmentManager = getSupportFragmentManager();
        fragmentManager.beginTransaction()
                .replace(R.id.frame_body, productsDetailsFragment)
                .addToBackStack(ProductsFragment.TAG)
                .commit();
    }
}

File: app/src/main/java/com/yalonso/interview/presentation/products/ProductsActivity.kt
package com.yalonso.interview.presentation.products

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.FragmentManager
import com.yalonso.interview.R
import com.yalonso.interview.presentation.productDetails.ProducDetailsFragment
import com.yalonso.interview.presentation.products.ProductsFragment.Companion.newInstance

class ProductsActivity: AppCompatActivity() {

    private lateinit var productsFragment: ProductsFragment
    private lateinit var productsDetailsFragment: ProducDetailsFragment
    private lateinit var fragmentManager: FragmentManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.products_activity)
        showProductsFragment()
    }

    protected fun showProductsFragment() {
        productsFragment = newInstance()
        fragmentManager = supportFragmentManager
        fragmentManager.beginTransaction()
            .replace(R.id.frame_body, productsFragment)
            .addToBackStack(ProductsFragment.TAG)
            .commit()
    }

//    fun pushProductFragment(product: Product?) {
//        productsDetailsFragment = newInstance(product!!)
//        fragmentManager = supportFragmentManager
//        fragmentManager.beginTransaction()
//            .replace(R.id.frame_body, productsDetailsFragment)
//            .addToBackStack(ProductsFragment.TAG)
//            .commit()
//    }

}